### ISSUE TITLE
weirdness with marking notifications as read/unread

### ISSUE THREAD

**Issue by JChipault** [USGS-WiM/whispers#949]
Selecting and marking notifications as read or unread doesn't always perform as expected. I'm not quite sure what's going on.

On 3/12 in chrome, I got into this conundrum:
![image](https://user-images.githubusercontent.com/49287076/77473150-052c1400-6de3-11ea-8298-5c28666d839b.png)

And then today in firefox I got into the situation where the "select all" box had a checkmark but none of my 17 messages had a checkmark in front of them. Only one of my 17 messages is marked as read:
![image](https://user-images.githubusercontent.com/49287076/77473244-25f46980-6de3-11ea-9dfd-f84e4597ca51.png)

And then I select "Mark as Read" and get this. Fair enough.
![image](https://user-images.githubusercontent.com/49287076/77473418-7370d680-6de3-11ea-90db-3edfb62ebba1.png)

But then I select "Mark as Unread" and the one message that was read before remains marked as read even though it's not individually selected (that I can see): 
![image](https://user-images.githubusercontent.com/49287076/77473344-550adb00-6de3-11ea-827a-c11e7b2b352b.png)

**Comment by BlakeDraper** [comment]
Fixes made in code:

1) Ensuring that the select all checkbox only selects items on the currently visible page
2) clearing the selection after an action is taken (mark as read or as unread)

**Comment by BlakeDraper** [comment]
Published in v2.14.0.

**Comment by JChipault** [comment]
@BlakeDraper this is still acting kind of wonky. Here's what my screen looked like when I went to test this feature after just testing github 948 (firefox, v2.14.1):
![image](https://user-images.githubusercontent.com/49287076/78384414-c1ce6400-759f-11ea-9e94-cb81765982d6.png)

Then I clicked on the "select all" checkbox in the header row and got this:
![image](https://user-images.githubusercontent.com/49287076/78384497-e1fe2300-759f-11ea-9d0b-b2c8ae5116ba.png)

And then if I sort by source again I get:
![image](https://user-images.githubusercontent.com/49287076/78384749-5e910180-75a0-11ea-8f5d-627318ae4367.png)

Same thing happens in chrome. Did not try Edge.
